pkgs/by-name/op/opencl-clang/package.nix

```diff
@@ -2,58 +2,74 @@
   lib,
   stdenv,
   applyPatches,
+  callPackage,
+  buildPackages,
+  targetPackages,
   fetchFromGitHub,
   cmake,
   git,
-  llvmPackages_15,
   spirv-llvm-translator,
-  buildWithPatches ? true,
 }:
 
 let
+  llvmPkgs =
+    let
+      llvmPkgs = callPackage ./llvm {
+        buildLlvmTools = buildPackages.opencl-clang.llvmPkgs.tools;
+        targetLlvmLibraries = targetPackages.opencl-clang.llvmPkgs.libraries or llvmPkgs.libraries;
+        targetLlvm = targetPackages.opencl-clang.llvmPkgs.llvm or llvmPkgs.llvm;
+        version = "15.0.7";
```

Comment:
This should probably have an identifier to distinguish it from upstream LLVM and include the `opencl-clang` version. (Is it intended that 15.0.3 ≠ 15.0.7, though?)